lms/djangoapps/ccx/views.py

```diff
@@ -181,6 +187,11 @@ def create_ccx(request, course, ccx=None):
                 override_field_for_ccx(ccx, vertical, hidden, True)
 
     ccx_id = CCXLocator.from_course_locator(course.id, ccx.id)  # pylint: disable=no-member
+    with ccx_course(ccx_id) as course:
+        try:
+            allow_access(course, request.user, "ccx_coach")
+        except Role.DoesNotExist:
+            return HttpResponseBadRequest("Role does not exist.")
```

Comment:
When coach creates ccx for the first time, I am assigning him role "ccx_coach" on the ccx.
Now he has role =  "ccx_coach" on both master course and ccx. He can access dashboard from both of them.